src/Chapter3.hs

```diff
@@ -971,6 +1083,29 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Day =
+      Sunday
+    | Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    deriving (Enum, Show, Eq)
+
+isWeekend :: Day -> Bool
+isWeekend Sunday   = True
+isWeekend Saturday = True
+isWeekend _        = False
+
+nextDay :: Day -> Day
+nextDay day =
+    let nextDayIndex = (fromEnum day + 1) `mod` 7
+    in toEnum nextDayIndex
```

Comment:
Interesting solution with `Enum` instance 👍🏼 

You could also reuse `Bounded` typeclass here, with the succ function and a special case for the corner-case. It would be a tiny bit efficient, as here you make from/to enum.